weatherApp/Controllers/DetailWeatherViewController.swift

```diff
@@ -0,0 +1,57 @@
+//
+//  DetailWeatherViewController.swift
+//  weatherApp
+//
+//  Created by Dario Nikolic on 04/08/2020.
+//  Copyright © 2020 Dario Nikolic. All rights reserved.
+//
+
+import UIKit
+import Kingfisher
+
+class DetailWeatherViewController: UIViewController {
+    var currentWeather: CurrentWeather!
+    
+    @IBOutlet private weak var temp: UILabel!
+    @IBOutlet weak var weatherIcon: UIImageView!
+    @IBOutlet private weak var weatherDescription: UILabel!
+    @IBOutlet private weak var humidity: UILabel!
+    @IBOutlet private weak var pressure: UILabel!
+    @IBOutlet private weak var minTemp: UILabel!
+    @IBOutlet private weak var maxTemp: UILabel!
+    @IBOutlet private weak var feelsLike: UILabel!
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        
+        setupUiElements()
+    }
+    
+    override func viewWillLayoutSubviews() {
+        super.viewWillLayoutSubviews()
+
+        view.setGradientBackground(colorOne: Colors.grey, colorTwo: Colors.darkNavyBlue)
+    }
+    
+    //MARK: - UI elements setup
+    
+    private func setupUiElements(){
+        let forecast = currentWeather.forecast
```

Comment:
A couple of things:
1. `setupUiElements` is not a good name for what it is doing - I am expecting that it styles the elements but it is actually assigning values to UI elements.
2. This should not be called on `viewDidLoad` because then you have to be sure that `currentWeather` exists. You should update the UI only when data is set. You could do something like this:
```
var currentWeather: CurrentWeather! {
    didSet {
        updateUIElements()
    }
}
```
And then you assign values in the `func`.